tools/codegen/api/lazy.py

```diff
@@ -108,14 +132,15 @@ def __init__(self, func: FunctionSchema):
                           "tensor_options",
                           "post_tensor_options_kwarg_only",
                           "out"]:
-            if getattr(func.arguments, arg_field) is not None:
-                keyword_arg_types.extend([
-                    NamedCType(
-                        arg.name,
-                        process_ir_type(arg.type)) for arg in getattr(func.arguments, arg_field)])
+            curr_args = getattr(func.arguments, arg_field)
+            if curr_args is not None:
+                if isinstance(curr_args, TensorOptionsArguments):
+                    curr_args = curr_args.all()
+                keyword_arg_types.extend([NamedCType(arg.name, process_ir_type(arg.type)) for arg in curr_args])
```

Comment:
Basic question - it looks like right now, `LazyIrSchema` has separate fields for positional and keyword arguments. Is that info used meaningfully anywhere?

If not, I think you could simplify a bunch of this logic into something like:
```
self.keyword_arg_types = [NamedCType(arg.name, process_ir_type(arg.type)) for ag in func.schema_order_arguments()]
```

That would help you avoid having to special-case `TensorOptionsArguments`, `SelfArgument`, and have to explicitly list out the `Arguments` field names (which I think would cause LTC codegen to break in pretty subtle ways if we changed the names of those fields in core codegen)

Reply:
if you look at `lazy_ts_lowering.py` you'll see how we generate different code for kw args vs positional args.  I couldn't think of a cleaner way at the time but I agree it's a bit complex.